packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -48,27 +48,190 @@ export function Comments( {
 	showCommentBoard,
 	setShowCommentBoard,
 } ) {
-	const { blockCommentId } = useSelect( ( select ) => {
-		const { getBlockAttributes, getSelectedBlockClientId } =
-			select( blockEditorStore );
-		const _clientId = getSelectedBlockClientId();
-
-		return {
-			blockCommentId: _clientId
-				? getBlockAttributes( _clientId )?.blockCommentId
-				: null,
-		};
-	}, [] );
+	const prevRef = useRef( null );
+	const retryRef = useRef( null );
 
-	const [ focusThread, setFocusThread ] = useState(
-		showCommentBoard && blockCommentId ? blockCommentId : null
+	const { blockCommentId, blocks, selectedBlockClientId } = useSelect(
+		( select ) => {
+			const { getBlockAttributes, getSelectedBlockClientId, getBlocks } =
+				select( blockEditorStore );
+			const _clientId = getSelectedBlockClientId();
+
+			return {
+				blockCommentId: _clientId
+					? getBlockAttributes( _clientId )?.blockCommentId
+					: null,
+				blocks: getBlocks(),
+				selectedBlockClientId: _clientId,
+			};
+		},
+		[]
 	);
 
 	const clearThreadFocus = () => {
 		setFocusThread( null );
 		setShowCommentBoard( false );
 	};
 
+	const { selectBlock } = useDispatch( blockEditorStore );
+
+	const [ focusThread, setFocusThread ] = useState(
+		showCommentBoard && blockCommentId ? blockCommentId : null
+	);
+
+	const { toggleBlockHighlight } = useDispatch( blockEditorStore );
+
+	useEffect( () => {
+		const reset = () => {
+			if ( retryRef.current ) {
+				clearTimeout( retryRef.current );
+				retryRef.current = null;
+			}
+			if ( prevRef.current ) {
+				toggleBlockHighlight( prevRef.current, false );
+				prevRef.current = null;
+			}
+		};
+
+		reset();
+		if ( ! selectedBlockClientId ) {
+			return;
+		}
+
+		try {
+			toggleBlockHighlight( selectedBlockClientId, true );
+			prevRef.current = selectedBlockClientId;
+
+			// Scroll block to center
+			const blockElement = document.querySelector(
+				`[data-block="${ selectedBlockClientId }"]`
+			);
```

Comment:
Let's avoid this approach. `document.querySelector` won't work because the block is inside the iframe editor instance. This code means we select the list view item, not the block.